include/rapidcheck/state/gen/ParallelCommands.hpp

```diff
@@ -0,0 +1,341 @@
+#pragma once
+
+#include "rapidcheck/state/gen/Commands.h"
+
+namespace rc {
+namespace state {
+namespace gen {
+
+namespace detail {
+template <typename Cmd, typename GenFunc>
+class ParallelCommandsGen {
+public:
+  using CmdSP = std::shared_ptr<const Cmd>;
+  using Model = typename Cmd::Model;
+  using Sut = typename Cmd::Sut;
+
+  template <typename ModelArg, typename GenFuncArg>
+  ParallelCommandsGen(ModelArg &&initialState, GenFuncArg &&genFunc)
+      : m_initialState(std::forward<ModelArg>(initialState))
+      , m_genFunc(std::forward<GenFuncArg>(genFunc)) {}
+
+  Shrinkable<ParallelCommands<Cmd>> operator()(const Random &random,
+                                               int size) const {
+    return generateCommands(random, size);
+  }
+
+private:
+  struct CommandEntry {
+    CommandEntry(Random &&aRandom, Shrinkable<CmdSP> &&aShrinkable)
+        : random(std::move(aRandom))
+        , shrinkable(std::move(aShrinkable)) {}
+
```

Comment:
I think the state should be there so unless you have something else you want to do before I'd say go ahead with it.

I don't really know what there is left to do?